tests/GridFS/SpecificationTests.php

```diff
@@ -0,0 +1,255 @@
+                                $col->deleteMany($delete['q']);
+                            }
+                        }
+        );
+    }
+    /**
+     *@dataProvider provideSpecificationTests
+     */
+    public function testSpecificationTests($testJson)
+    {
+        foreach ($testJson['tests'] as $test) {
+            $this->initializeDatabases($testJson['data'], $test);
+
+            if(isset($test['act']['arguments']['options'])){
+                $options = $test['act']['arguments']['options'];
+            } else {
+                $options =[];
+            }
+            $this->bucket = new \MongoDB\GridFS\Bucket($this->manager, $this->getDatabaseName(), $this->fixTypes($options,false));
+            $this->bucketReadWriter = new \MongoDB\GridFS\BucketReadWriter($this->bucket);
+            $func = $test['act']['operation'] . "Command";
+            $error = null;
+            try {
+                $result = $this->$func($test['act']['arguments']);
+            } catch(\MongoDB\Exception\Exception $e) {
+                $error = $e;
+            }
+
+            $errors = ['FileNotFound' =>  '\MongoDB\Exception\GridFSFileNotFoundException',
+                        'ChunkIsMissing' => '\MongoDB\Exception\GridFSCorruptFileException',
+                        'ExtraChunk' => '\MongoDB\Exception\GridFSCorruptFileException',
+                        'ChunkIsWrongSize' => '\MongoDB\Exception\GridFSCorruptFileException',
+                        'RevisionNotFound' => '\MongoDB\Exception\GridFSFileNotFoundException'
+                ];
+
+            if (!isset($test['assert']['error'])) {
+                $this->assertNull($error);
+
+            } else {
+                $shouldError = $test['assert']['error'];
+                $this->assertTrue($error instanceof $errors[$shouldError]);
+            }
+            $fixedAssert = $this->fixTypes($test['assert'], false);
+            if (isset($test['assert']['result'])) {
+                $testResult = $test['assert']['result'];
+                if ($testResult == "&result") {
+                    $test['assert']['result'] = $result;
+                }
+                if ($testResult == "void") {
+                    $fixedAssert['result'] = null;
+                }
+                $this->assertEquals($result, $fixedAssert['result']);
+            }
+            if (isset($test['assert']['data'])) {
+                $this->runCommands($fixedAssert['data'], $result);
+                $this->collectionsEqual($this->collections['expected.files'],$this->bucket->getFilesCollection());
+                if(isset($this->collections['expected.chunks'])) {
+                    $this->collectionsEqual($this->collections['expected.chunks'],$this->bucket->getChunksCollection());
+                }
+            }
+        }
+    }
+
+    public function provideSpecificationTests()
+    {
+        $testPath=getcwd().'/tests/GridFS/Specification/tests/download_by_name.json';
```

Comment:
It'd be more portable to use the `__DIR__` constant here and give the path relative to this file.